libraries/package-extractor/src/PackageExtractor.ts

```diff
@@ -483,8 +499,13 @@ export class PackageExtractor {
           path.join(packageJsonRealFolderPath, 'package.json')
         );
 
+        const targetSubspace: IExtractorSubspace | undefined = subspaces?.find(
```

Comment:
Here's a couple examples of how Rush handles these problems:

- [LookupByPath.ts](https://github.com/microsoft/rushstack/blob/e43b2f686ed191dbda67eb904a199120cb82e3e0/libraries/rush-lib/src/logic/LookupByPath.ts#L50) constructs an efficient lookup tree; OR
- [PackageJsonLookup](https://github.com/microsoft/rushstack/blob/e43b2f686ed191dbda67eb904a199120cb82e3e0/libraries/node-core-library/src/PackageJsonLookup.ts) simply caches the results.

@EscapeB @william2958 Because PR #4655 is an important fix, we don't need to address the problem in this PR. But I think we should consider optimizing these lookups because `O(n^2)` loops can and do cause noticeable slowness.

Reply:
OK，will try to improve and that in later PR.